### ISSUE TITLE
Downloading some episodes cannot be played

### ISSUE THREAD

**Issue by ltsj** [crunchy-labs/crunchy-cli#32]
When I download episode 6, it can't be played after the download is complete. But I downloaded episode 5 again and it played fine.

`./crunchy archive -l ja-JP https://beta.crunchyroll.com/series/G5PHNM403/the-dawn-of-the-witch[S01E06] -v -g 5`

Download process:

```
[debug] Entering season 1, episode 6 with 0 additional formats
[debug] Set up signal catcher
[info] Downloading episode `Knowledge of Ignorance` to `Knowledge of Ignorance.mkv` (merging audio for additional formats)
[info]  Episode: S01E06
[info]  Audio: ja-JP
[info]  Subtitle: 
[info]  Resolution: 1920x1080px
[info]  FPS: 23.97


[debug] Merging segments
[debug] Downloaded 'ja-JP' video
[debug] Downloaded 'ar-SA' subtitles
[debug] Downloaded 'en-US' subtitles
[debug] Downloaded 'fr-FR' subtitles
[debug] Downloaded 'de-DE' subtitles
[debug] Downloaded 'it-IT' subtitles
[debug] Downloaded 'pt-BR' subtitles
[debug] Downloaded 'ru-RU' subtitles
[debug] Downloaded 'es-419' subtitles
[debug] Downloaded 'es-ES' subtitles
[debug] FFmpeg merge command: ffmpeg -y -i "/tmp/Knowledge of Ignorance.mkv_ja-JP_video_4228809968.ts" -i "/tmp/Knowledge of Ignorance.mkv_ar-SA_subtitle_2246873797.ass" -i "/tmp/Knowledge of Ignorance.mkv_en-US_subtitle_96231048.ass" -i "/tmp/Knowledge of Ignorance.mkv_fr-FR_subtitle_3549463093.ass" -i "/tmp/Knowledge of Ignorance.mkv_de-DE_subtitle_3645851011.ass" -i "/tmp/Knowledge of Ignorance.mkv_it-IT_subtitle_2836418908.ass" -i "/tmp/Knowledge of Ignorance.mkv_pt-BR_subtitle_2467827141.ass" -i "/tmp/Knowledge of Ignorance.mkv_ru-RU_subtitle_3717983405.ass" -i "/tmp/Knowledge of Ignorance.mkv_es-419_subtitle_1758587536.ass" -i "/tmp/Knowledge of Ignorance.mkv_es-ES_subtitle_3931097704.ass" -map 0 -map 1 -map 2 -map 3 -map 4 -map 5 -map 6 -map 7 -map 8 -map 9 -metadata:s:v:0 language="ja-JP" -metadata:s:v:0 title="Japanese" -metadata:s:a:0 language="ja-JP" -metadata:s:a:0 title="Japanese" -metadata:s:s:0 language="ar-SA" -metadata:s:s:0 title="Arabic" -metadata:s:s:1 language="en-US" -metadata:s:s:1 title="English (US)" -metadata:s:s:2 language="fr-FR" -metadata:s:s:2 title="French" -metadata:s:s:3 language="de-DE" -metadata:s:s:3 title="German" -metadata:s:s:4 language="it-IT" -metadata:s:s:4 title="Italian" -metadata:s:s:5 language="pt-BR" -metadata:s:s:5 title="Portuguese (Brazil)" -metadata:s:s:6 language="ru-RU" -metadata:s:s:6 title="Russian" -metadata:s:s:7 language="es-419" -metadata:s:s:7 title="Spanish (Latin America)" -metadata:s:s:8 language="es-ES" -metadata:s:s:8 title="Spanish (Spain)" -disposition:s:0 0 -c copy -f matroska /tmp/3676206230
[debug] Download finished
[debug] Stopped signal catcher
```
And the file that cannot be played normally contains a German subtitle with an abnormal time，When I remove this subtitle file and remix the stream, the video can be played normally.

```
ID                                       : 6
Format                                   : ASS
Codec ID                                 : S_TEXT/ASS
Codec ID/Info                            : Advanced Sub Station Alpha
Duration                                 : 3 时 59 分
Compression mode                         : 无损
Title                                    : German
Language                                 : German (DE)
Default                                  : 否
Forced                                   : 否
```

This is the 5th episode, it can be played normally after it is downloaded, but the 6th episode cannot.

`./crunchy archive -l ja-JP https://beta.crunchyroll.com/series/G5PHNM403/the-dawn-of-the-witch[S01E05] -v -g 5`

```
[debug] Entering season 1, episode 5 with 0 additional formats
[debug] Set up signal catcher
[info] Downloading episode `I Decide Who's Worthy` to `I Decide Who's Worthy.mkv` (merging audio for additional formats)
[info]  Episode: S01E05
[info]  Audio: ja-JP
[info]  Subtitle: 
[info]  Resolution: 1920x1080px
[info]  FPS: 23.97



[debug] Merging segments
[debug] Downloaded 'ja-JP' video
[debug] Downloaded 'ar-SA' subtitles
[debug] Downloaded 'en-US' subtitles
[debug] Downloaded 'fr-FR' subtitles
[debug] Downloaded 'de-DE' subtitles
[debug] Downloaded 'it-IT' subtitles
[debug] Downloaded 'pt-BR' subtitles
[debug] Downloaded 'ru-RU' subtitles
[debug] Downloaded 'es-419' subtitles
[debug] Downloaded 'es-ES' subtitles
[debug] FFmpeg merge command: ffmpeg -y -i "/tmp/I Decide Who's Worthy.mkv_ja-JP_video_3629755215.ts" -i "/tmp/I Decide Who's Worthy.mkv_ar-SA_subtitle_594230327.ass" -i "/tmp/I Decide Who's Worthy.mkv_en-US_subtitle_775737117.ass" -i "/tmp/I Decide Who's Worthy.mkv_fr-FR_subtitle_2237748135.ass" -i "/tmp/I Decide Who's Worthy.mkv_de-DE_subtitle_2171227641.ass" -i "/tmp/I Decide Who's Worthy.mkv_it-IT_subtitle_1034428868.ass" -i "/tmp/I Decide Who's Worthy.mkv_pt-BR_subtitle_1369825909.ass" -i "/tmp/I Decide Who's Worthy.mkv_ru-RU_subtitle_2177413224.ass" -i "/tmp/I Decide Who's Worthy.mkv_es-419_subtitle_1392795710.ass" -i "/tmp/I Decide Who's Worthy.mkv_es-ES_subtitle_3336197141.ass" -map 0 -map 1 -map 2 -map 3 -map 4 -map 5 -map 6 -map 7 -map 8 -map 9 -metadata:s:v:0 language="ja-JP" -metadata:s:v:0 title="Japanese" -metadata:s:a:0 language="ja-JP" -metadata:s:a:0 title="Japanese" -metadata:s:s:0 language="ar-SA" -metadata:s:s:0 title="Arabic" -metadata:s:s:1 language="en-US" -metadata:s:s:1 title="English (US)" -metadata:s:s:2 language="fr-FR" -metadata:s:s:2 title="French" -metadata:s:s:3 language="de-DE" -metadata:s:s:3 title="German" -metadata:s:s:4 language="it-IT" -metadata:s:s:4 title="Italian" -metadata:s:s:5 language="pt-BR" -metadata:s:s:5 title="Portuguese (Brazil)" -metadata:s:s:6 language="ru-RU" -metadata:s:s:6 title="Russian" -metadata:s:s:7 language="es-419" -metadata:s:s:7 title="Spanish (Latin America)" -metadata:s:s:8 language="es-ES" -metadata:s:s:8 title="Spanish (Spain)" -disposition:s:0 0 -c copy -f matroska /tmp/458289425
[debug] Download finished
[debug] Stopped signal catcher

```

**Comment by ltsj** [comment]
`./crunchy download -r best https://beta.crunchyroll.com/series/G5PHNM403/the-dawn-of-the-witch[S01E06] -g 5`

I tried downloading the .ts problem and it was no problem. Seems to be a problem converting to .mkv

**Comment by ltsj** [comment]
I tested again when I use:

`./crunchy download -o mkv -r best https://beta.crunchyroll.com/series/G5PHNM403/the-dawn-of-the-witch[S01E06] -g 5 -v`

At this time, the downloaded mkv file can be played normally.

**Comment by ByteDream** [comment]
That's a weird issue, normally this shouldn't happen. Maybe there was a issue while downloading the subtitle file which caused this.

**Comment by ltsj** [comment]
The problem reappeared when downloading "[Case Closed (Detective Conan)](https://beta.crunchyroll.com/series/G6JQVM3ER/case-closed-detective-conan)"

Remix and remove all subtitles and the problem goes away

Just give feedback on the problems in use

**Comment by ByteDream** [comment]
For me, the episode 6 video somehow extends to 4 hours length, but only around 24 minutes (the actual length the episode) can be played, after this my video player (vlc / mpv) stops the video entirely. If I remove the German subtitle the video has the normal 24 minutes length. Mysterious.

**Comment by ByteDream** [comment]
In the `.ass` file which contains the German subtitles, one entry has the end-time of around 4 hours. This is the cause of the extended video length. Once again a design decision I can not understand ¯\\_(ツ)_/¯.
![image](https://user-images.githubusercontent.com/63594396/188275324-f987664b-f9c8-472d-baf2-eb659d8b2ed8.png)

**Comment by ByteDream** [comment]
@ltsj which video player are you using? When I download it on Windows 10 it can be played normally with the built-in video player, at least until the addressed subtitle bug occurs

**Comment by ltsj** [comment]
I use the latest version of PotPlayer

**Comment by ByteDream** [comment]
Fixed in release [v2.3.1](https://github.com/crunchy-labs/crunchy-cli/releases/tag/v2.3.1)
